src/dotnet/App.Maui/Platforms/Android/AndroidWebViewClient.cs

```diff
@@ -31,13 +31,25 @@ public override bool OnRenderProcessGone(WebView? view, RenderProcessGoneDetail?
         var didCrash = detail?.DidCrash() == true;
         var details = $"DidCrash: {didCrash}, RendererPriorityAtExit: {detail?.RendererPriorityAtExit()}, {detail}";
         log.LogWarning("OnRenderProcessGone: {Details}", details);
-        var markedDead = view.IsNotNull()
+
+        if (view.IsNotNull()
             && MauiWebView.Current is { } mauiWebView
             && mauiWebView.AndroidWebView.IfNotNull() is { } androidWebView
-            && ReferenceEquals(androidWebView, view)
-            && mauiWebView.MarkDead();
-        if (didCrash && markedDead) {
-            MainThread.BeginInvokeOnMainThread(() => MainPage.Current.RecreateWebView());
+            && ReferenceEquals(androidWebView, view)) {
+            if (mauiWebView.MarkDead()) {
+                androidWebView.ClearCache(false);
+                androidWebView.RemoveAllViews();
+                androidWebView.OnPause();
+                androidWebView.ClearHistory();
+                MainThread.BeginInvokeOnMainThread(() => {
+                    if (didCrash)
```

Comment:
Why do you need here different approaches to handle cases when render process has gone due to crash or not?

Reply:
Because it can be dropped due to memory pressure while the app is in background - and we will recreate it on getting foreground